File: momhelp/src/main/java/com/momhelp/service/DishService.java
package com.momhelp.service;

import com.momhelp.dto.DishDTO;
import java.util.List;

public interface DishService {

	// Get all dishes
	List<DishDTO> getAllDishes();

	// Get dish by ID
	DishDTO getDishById(Long id);

	// Get dishes by vegetable ID
	List<DishDTO> getDishesByVegetableId(Long vegetableId);

	// Get dishes by vegetable name
	List<DishDTO> getDishesByVegetableName(String vegetableName);

	// Get dishes for available vegetables (not spoiled, not expired)
	List<DishDTO> getDishesForAvailableVegetables();

	// Get available dishes for a specific vegetable
	List<DishDTO> getAvailableDishesByVegetableId(Long vegetableId);

	// Get all vegetables that have dishes (for Choose Me section)
	List<String> getVegetablesWithDishes();
}

File: momhelp-frontend/momhelp-frontend/src/services/dishService.js
// import api from './api';

// const dishService = {
//   // Get all dishes
//   getAllDishes: () => {
//     return api.get('/dishes/all');
//   },

//   // Get dish by ID
//   getDishById: (id) => {
//     return api.get(`/dishes/${id}`);
//   },

//   // Get dishes by vegetable ID
//   getDishesByVegetableId: (vegetableId) => {
//     return api.get(`/dishes/by-vegetable/${vegetableId}`);
//   },

//   // Get available dishes for a vegetable (for Choose Me flow)
//   getAvailableDishesByVegetableId: (vegetableId) => {
//     return api.get(`/dishes/available/${vegetableId}`);
//   },

//   // Get dishes for available vegetables
//   getDishesForAvailableVegetables: () => {
//     return api.get('/dishes/available');
//   },

//   // Get vegetables that have dishes
//   getVegetablesWithDishes: () => {
//     return api.get('/dishes/vegetables-with-dishes');
//   },

//   // Choose Me flow: Get dishes for vegetable
//   getChooseMeDishes: (vegetableId) => {
//     return api.get(`/choose-me/dishes/${vegetableId}`);
//   }

//   // REMOVED: getChooseMeRecipe from here (it belongs in recipeService)
// };

// export default dishService;


import api from './api';

const dishService = {
  // Get all dishes
  getAllDishes: () => {
    return api.get('/dishes/all');
  },

  // Get dish by ID
  getDishById: (id) => {
    return api.get(`/dishes/${id}`);
  },

  // Get dishes by vegetable ID
  getDishesByVegetableId: (vegetableId) => {
    return api.get(`/dishes/by-vegetable/${vegetableId}`);
  },

  // Get available dishes for a vegetable (for Choose Me flow)
  getAvailableDishesByVegetableId: (vegetableId) => {
    return api.get(`/dishes/available/${vegetableId}`);
  },

  // Get dishes for available vegetables
  getDishesForAvailableVegetables: () => {
    return api.get('/dishes/available');
  },

  // Get vegetables that have dishes
  getVegetablesWithDishes: () => {
    return api.get('/dishes/vegetables-with-dishes');
  },

  // ✅ NEW: Choose Me flow with language support
  getChooseMeDishes: (vegetableId, language = 'en') => {
    return api.get(`/choose-me/dishes/${vegetableId}`, {
      params: { language }
    });
  }
};

export default dishService;